src/shared/bootspec.c

```diff
@@ -776,9 +776,11 @@ int boot_entries_augment_from_loader(BootConfig *config, bool only_auto) {
                                 break;
                         }
 
+#if ENABLE_EFI
                 p = efi_variable_path(EFI_VENDOR_LOADER, "LoaderEntries");
                 if (!p)
                         return log_oom();
+#endif
```

Comment:
this doesn't look right. This should be handle the in efivars.h header file, like we do for the rest of EFI calls. Most likely efi_variable_path() should simply not be conditioned at all.

That said, I figure we can conditionalize the whole body of `boot_entries_augment_from_loader()` if EFI is off, since it has no effect then.